src/main/webapp/js/filemanager.js

```diff
@@ -1237,111 +1240,260 @@ function fileManager(_initialData, _$beaconTable, _startURI, _folderPath,
+        response(results.slice(0, 10));
+      },
+      minLength: 2
+    });
 
-        var iconAnchor = $(event.currentTarget).find("a")[0];
+    $("#writeGroup").keyup(function ()
+    {
+      togglePromptError("#writeGroupDiv", "#writeGroupLabel",lg.WRITE_GROUP, "off");
+    });
 
-        var checkboxState = "";
-        if (iconAnchor.getAttribute("readable") === "true")
-        {
-          checkboxState = "checked";
-        }
+  };
 
-        // Add form body
-        var msg =  '<label class="prompt-label custom-control custom-checkbox">' +
-           '<span class="custom-control-indicator"></span>' +
-           '<span class="custom-control-description">Public read permission</span>' +
-           '</label>' +
-           '<input type="checkbox" class="form-control prompt-checkbox" id="publicPermission" name="publicPermission" ' + checkboxState + ' >' +
-           '<input type="text" class="hidden" name="itemPath" id="itemPath" value="' + iconAnchor.getAttribute("path") +
-            '">';
-
-        var btns = [];
-        btns.push
-        ({
-          "title": lg.save,
-          "value": true,
-          "classes": "btn btn-primary"
-        });
-        btns.push
-        ({
-            "title": lg.cancel,
-            "value": false,
-            "classes": "btn btn-default"
-        });
+  // Submit function for $.prompt instance
+  var handleEditPermissions = function (event, value, message, formVals)
+  {
+    if (value === true)
+    {
+      var readGroupValid = false;
+      var writeGroupValid = false;
+
+      if ((typeof(formVals["readGroup"]) === "undefined") ||
+          (formVals["readGroup"] === "") ||
+          ($.inArray(formVals["readGroup"], autoCompleteList) >= 0))
+      {
+        readGroupValid = true;
+      }
 
 
-        $.prompt(msg,
+      if ((typeof(formVals["writeGroup"]) === "undefined") ||
+          (formVals["writeGroup"] === "") ||
+          ($.inArray(formVals["writeGroup"], autoCompleteList) >= 0))
+      {
+        writeGroupValid = true;
+      }
+
+
+      if ((writeGroupValid === true) && (readGroupValid === true))
+      {
+        // var publicCheckbox = formVals['publicPermission'];
+        var itemPath = formVals['itemPath'];
+
+        var url = contextPath + config.options.itemConnector + itemPath;
+
+        var dataStr = JSON.stringify(formVals);
+        $.ajax(
+        {
+          url: url,
+          method: "POST",
+          contentType: "application/json",
+          data: dataStr,
+          statusCode:
+          {
+            201: function () {
+              $.prompt(lg.successful_edit,
+                  {
+                    submit: refreshPage
+                  });
+            },
+            204: function ()
+            {
+              $.prompt(lg.successful_edit,
+                  {
+                    submit: refreshPage
+                  });
+            },
+            400: function ()
```

Comment:
Grant it, a 400 should never happen, but we might want to handle the message at any rate.  Something about, 'Invalid entry'.  Also, when does a 204 ever happen?